Extensions/Multiplayer/multiplayerobjectruntimebehavior.ts

```diff
@@ -0,0 +1,639 @@
+      if (
+        this.playerNumber !== 0 && // Host is always connected.
+        !gdjs.multiplayerMessageManager.isPlayerConnected(this.playerNumber)
+      ) {
+        logger.info(
+          `Player number ${this.playerNumber} does not exist in the lobby at the moment. Marking the object for destruction.`
+        );
+        this._shouldDeleteObject = true;
+        return;
+      }
+
+      // When a synchronized object is created, we assume it will be assigned a networkId quickly if:
+      // - It is a new object created by the current player. -> will be assigned a networkId when sending the update message.
+      // - It is an object created by another player. -> will be assigned a networkId when receiving the update message.
+      // There is a small risk that the object is created by us after we receive an update message from the host,
+      // ending up with 2 objects created, one with a networkId (from the host) and one without (from us).
+      // To handle this case and avoid having an object not synchronized, we set a timeout to destroy the object
+      // if it has not been assigned a networkId after a short delay.
+      this._destroyInstanceTimeoutId = setTimeout(() => {
+        if (!owner.networkId) {
+          logger.info(
+            `Object ${owner.getName()} has not been assigned a networkId after a short delay, destroying it.`
+          );
+          owner.deleteFromScene(instanceContainer);
+        }
+      }, this._timeBeforeDestroyingObjectWithoutNetworkIdInMs);
+
+      logger.info(
+        'Created MultiplayerObjectRuntimeBehavior object with name ' +
+          owner.getName()
+      );
+    }
+
+    sendDataToPeersWithIncreasedClock(messageName: string, data: Object) {
+      this._clock++;
+      data['_clock'] = this._clock;
+      const connectedPeerIds = gdjs.evtTools.p2p.getAllPeers();
+      for (const peerId of connectedPeerIds) {
+        gdjs.multiplayerMessageManager.sendDataTo(peerId, messageName, data);
+      }
+    }
+
+    isOwnerAsPlayerOrHost() {
+      const currentPlayerNumber = gdjs.multiplayer.getPlayerNumber();
+
+      const isOwnerOfObject =
+        currentPlayerNumber === this.playerNumber || // Player as owner.
+        (currentPlayerNumber === 1 && this.playerNumber === 0); // Host as owner.
+
+      return isOwnerOfObject;
+    }
+
+    hasObjectBeenSyncedWithinMaxRate() {
+      return (
+        this._getTimeNow() - this._lastObjectSyncTimestamp <
+        1000 / this._objectMaxTickRate
+      );
+    }
+
+    hasObjectBasicInfoBeenSyncedRecently() {
+      return (
+        this._getTimeNow() - this._lastBasicObjectSyncTimestamp <
+        1000 / this._objectBasicInfoTickRate
+      );
+    }
+
+    haveVariablesBeenSyncedRecently() {
+      return (
+        this._getTimeNow() - this._lastVariablesSyncTimestamp <
+        1000 / this._variablesTickRate
+      );
+    }
+
+    haveEffectsBeenSyncedRecently() {
```

Comment:
Nit: probably good to mark these as private + underscore them.

Reply:
done